### PR TITLE
Add TLS certificate e2e test

### PR DESCRIPTION
**Pull request by astefanutti**
This PR adds e2e tests covering TLS certificate issuance for Ingresses.

It introduces support for CA Issuer as a pre-requisite, but it's already capable of running with Let's Encrypt as cert-manager issuer.

This PR is a pre-requisite for #95.

This also enables parallel execution of the e2e test suite:

```
go test -timeout 60m -v ./e2e -tags=e2e
=== RUN   TestIngress
=== PAUSE TestIngress
=== RUN   TestTLS
=== PAUSE TestTLS
=== CONT  TestIngress
=== CONT  TestTLS
--- PASS: TestTLS (53.65s)
--- PASS: TestIngress (54.38s)
PASS
ok  	github.com/kuadrant/kcp-glbc/e2e	54.397s
```

vs:

```
go test -timeout 60m -v ./e2e -tags=e2e
=== RUN   TestIngress
--- PASS: TestIngress (54.42s)
=== RUN   TestTLS
--- PASS: TestTLS (59.14s)
PASS
ok  	github.com/kuadrant/kcp-glbc/e2e	113.576s
```

### CODE REVIEW

File: e2e/support/ingress.go [hunk 1]
```diff
@@ -43,6 +57,10 @@ func LoadBalancerIngresses(ingress *networkingv1.Ingress) []corev1.LoadBalancerI
 	return ingress.Status.LoadBalancer.Ingress
 }
 
+func IngressTLS(ingress *networkingv1.Ingress) []networkingv1.IngressTLS {
```
Comment (maleck13) [inline e2e/support/ingress.go:60]: Do we really need this function?

Comment (astefanutti) [reply e2e/support/ingress.go:60]: It's used for convenience to write `test.Expect(ingress).To(WithTransform(IngressTLS, ConsistOf(...)`, e.g., here:

https://github.com/Kuadrant/kcp-glbc/pull/114/files#diff-31685f841fd745a5ee0aa4959add25071cc2403a974b91fe9083d1b0eaacff73R111

**Review by maleck13 — COMMENTED** [review]
The changes generally look good. One concern I have is whether the self signed provider has a use outside of the tests? Do you see it being useful for local development also? I am wondering if we want to expose a metric to allow us to see which provider is configured that way you could potentially alert on bad configuration or bug where the wrong provider was configured?

**Comment by astefanutti** [comment]
> The changes generally look good. One concern I have is whether the self signed provider has a use outside of the tests? Do you see it being useful for local development also?

I think it's primarily useful for local development, as a better replacement over the existing "fake" provider, and homogenises TLS certificates management, e.g., for monitoring.

> I am wondering if we want to expose a metric to allow us to see which provider is configured that way you could potentially alert on bad configuration or bug where the wrong provider was configured?

The TLS certificate metrics have an `issuer` dimension, which can be used to create specific alerts that would check the configured issuer is the expected one for the environment.

**Pull request merged**